packages/block-editor/src/components/media-upload/index.native.js

```diff
@@ -137,15 +143,27 @@ export class MediaUpload extends Component {
 		const {
 			allowedTypes = [],
 			__experimentalOnlyMediaLibrary,
+			isAudioBlockMediaUploadEnabled,
 		} = this.props;
 
 		return this.getAllSources()
 			.filter( ( source ) => {
-				return __experimentalOnlyMediaLibrary
-					? source.mediaLibrary
-					: allowedTypes.some( ( allowedType ) =>
+				if ( __experimentalOnlyMediaLibrary ) {
+					return source.mediaLibrary;
+				} else if (
+					allowedTypes.every(
+						( allowedType ) =>
+							allowedType === MEDIA_TYPE_AUDIO &&
 							source.types.includes( allowedType )
-					  );
+					) &&
+					source.id !== URL_MEDIA_SOURCE
+				) {
+					return isAudioBlockMediaUploadEnabled === true;
+				}
+
+				return allowedTypes.some( ( allowedType ) =>
+					source.types.includes( allowedType )
+				);
 			} )
```

Comment:
Let me know if the logic here is clear or if it needs comments for more clarity.

Reply:
Thanks for simplifying it, now it's quite clear 💯 .